plans/phase-1/multigrescluster-cr-definition.md

```diff
@@ -0,0 +1,713 @@
+  * Illustrate the relationship between the CRs and the Kubernetes resources they generate to ensure a robust management strategy.
+  * Codify the operator's scope of responsibilities and behaviors within the `MultigresCluster` Custom Resource Definition.
+
+-----
+
+## Non-Goals
+
+  * To provide an exhaustive definition of every subordinate Kubernetes resource created by the operator.
+  * To include the final, formal Custom Resource Definition (CRD) schemas in this document.
+  * The feature set in this document will be implicit in the design of the CRs but not the main point of discussion.
+
+-----
+
+## Proposal
+
+The core of the operator's API is the `MultigresCluster` Custom Resource. This resource provides a single, unified interface for configuring an entire Multigres cluster. The operator will reconcile this top-level CR by creating and managing a series of more granular child resources for components like cells, shards, and topology servers.
+
+While the primary workflow involves defining the entire topology within the `MultigresCluster` CR, the design also accommodates advanced use cases. Users will have the option to provision and manage child resources independently by setting a `managed: false` flag, allowing for greater flexibility and integration with existing infrastructure.
+
+
+### User Stories
+
+As a platform operator, I want to define my entire multi-cell, sharded Multigres topology in a single YAML file and apply it with `kubectl`. The system should then automatically provision all the necessary components, connect them, and bring the database cluster online without further manual intervention.
+
+### Operator Design and Architecture
+
+The operator's core responsibility is to translate the high-level state defined in the `MultigresCluster` CR into a set of child CRs. Each child CR represents a distinct component of the cluster, such as a cell or a database shard. The controllers for these child CRs are then responsible for creating and managing the low-level Kubernetes resources (like `StatefulSets`, `Deployments`, and `Services`) required to run the system.
+
+### API Definition (`MultigresCluster` CRD)
+
+The user interacts with the operator primarily through the `MultigresCluster` custom resource. This single manifest serves as the source of truth for the entire cluster. 
+
+> Please note that this sample does not contain the tablegroups or shard definitions, which are discussed later in this document.
+
+```yaml
+apiVersion: [multigres.com/v1alpha1](https://multigres.com/v1alpha1)
+kind: MultigresCluster
+metadata:
+  name: example-multigres-cluster
+  namespace: multigres
+spec:
+  # globalTopoServer defines the topology server for the entire cluster.
+  # Required.
+  globalTopoServer:
+    # If 'managed' is true, the operator will create and manage the global EtcdLockserver CR.
```

Comment:
what's a `EtcdLockserver`?

Reply:
Good catch! It's a typo, I have changed it.